independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/ConfigService.java

```diff
@@ -0,0 +1,219 @@
+package io.quarkus.analytics;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+
+import io.quarkus.analytics.config.FileLocations;
+import io.quarkus.analytics.config.GroupIdFilter;
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.LocalConfig;
+import io.quarkus.analytics.dto.config.NoopRemoteConfig;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.rest.ConfigClient;
+import io.quarkus.analytics.util.FileUtils;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Decided the build analytics behaviour. Retrieves, stores and provides the configuration.
+ */
+public class ConfigService {
+    public static final String QUARKUS_ANALYTICS_DISABLED_LOCAL_PROP = "quarkus.analytics.disabled";
+    public static final String QUARKUS_ANALYTICS_PROMPT_TIMEOUT = "quarkus.analytics.prompt.timeout";
+    private static final String NEW_LINE = System.lineSeparator();
+    public static final String ACCEPTANCE_PROMPT = NEW_LINE
+            + "********************************************************************************************************"
+            + NEW_LINE
+            + "********************************************************************************************************"
+            + NEW_LINE
+            + "********************************************************************************************************"
+            + NEW_LINE
+            + "* Please help improve Quarkus!" + NEW_LINE
+            + "* Learn more: https://quarkus.io/usage-policy/" + NEW_LINE
```

Comment:
this should be https://quarkus.io/usage

Reply:
it was sugested that /usage is prime real-state and /usage-policy was more precise.

Reply:
-1 on making urls we print or use around quarkus.io boring/hard to read. same reason for why security-policy url update in https://github.com/quarkusio/quarkusio.github.io/pull/1665 got rejected.

Reply:
@paulrobinson @insectengine what's your take?

Reply:
I agree that we want easy to read and remember URLs. So for that reason, I do prefer https://quarkus.io/usage.

@insectengine was concerned that we might want to use /usage for something else in the future. What were you thinking of in particular James?

My concern is that 'usage' is a pretty vague term, and I'm not sure people will get what it is. However, I can't really think of anything better, so lets go with that unless anyone can think of anything better?

Reply:
the text describes what it does - you learn more.

Reply:
seems that https://tools.jboss.org/usage/  has been used for many years.
For consistency, will change back to /usage in the follow up PR

Reply:
On a follow up PR